common/djangoapps/student/views/dashboard.py

```diff
@@ -472,6 +472,24 @@ def get_dashboard_course_limit():
     return course_limit
 
 
+def check_for_unacknowledged_notices(context):
+    """
+    Checks the notices apps plugin context to see if there are any unacknowledged notices the user needs to take action
+    on. If so, build a redirect url to the first unack'd notice.
+    """
+    notice_url = None
+
+    if context.get("plugins", {}).get('notices'):
+        notices_context = context.get("plugins").get("notices")
+        if notices_context.get("unacknowledged_notices"):
+            notices = notices_context.get("unacknowledged_notices")
+            # We will only show one notice to the user one at a time. Build a redirect URL to the first notice in the
+            # list of unacknowledged notices.
+            notice_url = f"{settings.LMS_ROOT_URL}{notices[0]}?next={settings.LMS_ROOT_URL}/dashboard/"
```

Comment:
`reverse()` in the plugin should already come with a domain name right? It does in the web API.

Reply:
Hmm, should get the FQDN when using rest_frameworks `reverse` https://www.django-rest-framework.org/api-guide/reverse/#reverse. Wonder what's happening locally then.

Reply:
Ah yeah... You could probably pass in the request from the `existing_context` [here](https://github.com/edx/platform-plugin-notices/blob/main/notices/context_api.py#L17) if you wanted to avoid it, but I think this is fine. :+1: